pkg/orchestrator/orchestrator.go

```diff
@@ -1249,6 +1285,131 @@ func (s *ControllerServer) resourceAttach(c *gin.Context, resourceInfo *Resource
 	c.JSON(http.StatusOK, attachResourceResp)
 }
 
+// Returns the resource info and cloud client for a given resource
+func (s *ControllerServer) getResourceInfo(c *gin.Context) (resourceInfo *ResourceInfo, cloudClient string, err error) {
```

Comment:
How is this different than `getAndValidateResourceURLParams` (which also queries for the tag)?

Reply:
Realized I don't need this function, and the existing function works fine. Thanks!